### ISSUE TITLE
Portals4 MTL hang during finalize with flow control

### ISSUE THREAD

**Issue by ompiteam** [open-mpi/ompi#147]
A trivial MPI application sometimes causes the P4 MTL to hang during finalize.

```
#include <mpi.h> 
main (int argc, char *argv[]) { 

  MPI_Init (&argc, &argv); 
  MPI_Finalize (); 
  return(0); 
} 
```

When I run it on n processes (even on one node), (n-1) processes terminate, and one is running in pthread_join (): 

```
(gdb) where 
#0  0x00007f216325b03d in pthread_join () from /lib64/libpthread.so.0 
#1  0x00007f2160618580 in _PtlNIFini () from /home_nfs/devezep/DISTS/openmpi-default-bull/lib/libportals.so.4 
#2  0x00007f2160829820 in ompi_mtl_portals4_finalize (mtl=0x7f2160a34520) at mtl_portals4.c:164 
#3  0x00007f2160a377c2 in mca_pml_cm_component_fini () at pml_cm_component.c:182 
#4  0x00007f2164441269 in mca_pml_base_finalize () at base/pml_base_close.c:34 
#5  0x00007f21643b319f in ompi_mpi_finalize () at runtime/ompi_mpi_finalize.c:294 
#6  0x00007f21643d786d in PMPI_Finalize () at pfinalize.c:46 
#7  0x000000000040079b in main () 
```

**Comment by jsquyres** [comment]
@regrant @tkordenbrock Is this issue still relevant?  Will it be fixed for v2.0.0?  Or sometime in the 2.x series?  Please update the milestone as appropriate, or close this issue.  Thanks!

**Comment by tkordenbrock** [comment]
This was caused by a bug in the Portals4 Reference Implementation that has been fixed.
